src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java

```diff
@@ -488,15 +508,10 @@ public FormValidation doValidate(@QueryParameter(fixEmpty = true) String domain,
          * In a real deployment, often there are servers that don't respond or
          * otherwise broken, so try all the servers.
          */
-        public DirContext bind(String principalName, String password, List<SocketInfo> ldapServers) {
+        public DirContext bind(String principalName, String password, List<SocketInfo> ldapServers, Hashtable<String, String> props) {
             // in a AD forest, it'd be mighty nice to be able to login as "joe"
             // as opposed to "joe@europe",
             // but the bind operation doesn't appear to allow me to do so.
-            Hashtable<String, String> props = new Hashtable<String, String>();
-            props.put(Context.REFERRAL, "follow");
-            props.put("java.naming.ldap.attributes.binary","tokenGroups objectSid");
```

Comment:
So you still need to keep these - these are internal and not general properties to expose.
if they have been set then you could skip setting them rather than overwriting - but you need these defaults.